_docs/tasks/security/https-overlay.md

```diff
@@ -0,0 +1,216 @@
+---
+title: Mutual TLS over https services
+overview: This task shows how to enable mTLS on https services.
+
+order: 80
+
+layout: docs
+type: markdown
+---
+{% include home.html %}
+
+This task shows how Istio Mutual TLS works with https services. It includes: 1)
+Deploy an https service without Istio sidecar; 2) Deploy an https service with
+Istio with mTLS disabled; 3) Deploy an https service with mTLS enabled. For each
+deployment, connect to this service and verify it works.
+
+When Istio sidecar is deployed with an https service, the proxy automatically downgrades
+from L7 to L4 (no matter mTLS is enabled or not), which means it does not terminate the
+original https traffic. And this is the reason Istio can work on https services.
+
+## Before you begin
+
+* Set up Istio by following the instructions in the
+  [quick start]({{home}}/docs/setup/kubernetes/quick-start.html).
+  Note that authentication should be **disabled** at step 5 in the
+  [installation steps]({{home}}/docs/setup/kubernetes/quick-start.html#installation-steps).
+
+## Deploy an https service without Istio sidecar
+
+### Generate certificates
+
+You need to have openssl installed to run this command
+
+```bash
+$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/nginx.key -out /tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"
+```
+
+```bash
+$ kubectl create secret tls nginxsecret --key /tmp/nginx.key --cert /tmp/nginx.crt
+secret "nginxsecret" created
+```
+
+### Create an https service without Istio sidecar
+
+This subsection creates a nginx-based https service.
+
+First we need to create a configmap:
+
+```bash
+$ kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf
```

Comment:
Does this one need to be repeated for each demo? Maybe move the previous section

Reply:
Done.